src/app/components/task-form/task-form.component.ts

```diff
@@ -0,0 +1,72 @@
+import { formatDate } from '@angular/common';
+import { Component, Input, OnChanges, Output, SimpleChanges } from '@angular/core';
+import { FormBuilder, FormGroup, FormsModule, ReactiveFormsModule, Validators } from '@angular/forms';
+import { RouterModule } from '@angular/router';
+import { TaskFormValidationService } from 'src/app/sevices/taskFormValidation.service';
+import { FormErrorComponent } from '../form-error/form-error.component';
+import { TaskListComponent } from '../task-list/task-list.component';
+import { EventEmitter } from '@angular/core';
+import { formType } from 'src/app/constants/formType';
+import { taskStatus, taskType } from 'src/app/constants/taskConstants';
+import { Task } from 'src/app/models/task';
+
+@Component({
+  selector: 'app-task-form',
+  standalone: true,
+  imports: [FormErrorComponent, ReactiveFormsModule, FormsModule, TaskListComponent, RouterModule],
+  templateUrl: './task-form.component.html',
+  styleUrl: './task-form.component.scss'
+})
+export class TaskFormComponent implements OnChanges {
+  @Input() submitText: string = "Submit";
+  @Input() task?: Task | null = null;
```

Comment:
remove the ? since it's an input it will always be null if not passed from a parent component